search/search-client-connect/src/main/java/io/camunda/search/connect/os/OpensearchConnector.java

```diff
@@ -45,24 +48,44 @@ public final class OpensearchConnector {
 
   private final ConnectConfiguration configuration;
   private final ObjectMapper objectMapper;
+  private final PluginRepository pluginRepository;
+
+  private AwsCredentialsProvider credentialsProvider;
```

Comment:
🔧 this could be final, since it's either passed to or initialized in a constructor.

Reply:
Good point. Will make it!